filters/scheduler/fifo.go

```diff
@@ -0,0 +1,161 @@
+package scheduler
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+	log "github.com/sirupsen/logrus"
+	"github.com/zalando/skipper/filters"
+	"github.com/zalando/skipper/scheduler"
+)
+
+type (
+	fifoSpec   struct{}
+	fifoFilter struct {
+		config scheduler.Config
+		queue  *scheduler.FifoQueue
+	}
+)
+
+func NewFifo() filters.Spec {
+	return &fifoSpec{}
+}
+
+func (*fifoSpec) Name() string {
+	return filters.FifoName
+}
+
+// CreateFilter creates a fifoFilter, that will use a semaphore based
+// queue for handling requests to limit concurrency of a route. The first
+// parameter is maxConcurrency the second maxQueueSize and the third
+// timeout.
+func (s *fifoSpec) CreateFilter(args []interface{}) (filters.Filter, error) {
+	var f fifoFilter
+
+	// set defaults
+	f.config.MaxConcurrency = defaultMaxConcurreny
+	f.config.MaxQueueSize = defaultMaxQueueSize
+	f.config.Timeout = defaultTimeout
```

Comment:
Lets make all args mandatory as in the docs. There are only three and it forces user to think about the config.

Reply:
I just changed the code analog to lifo(), but I can make it mandatory.